skbio/stats/tests/test_composition.py

```diff
@@ -0,0 +1,192 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from __future__ import absolute_import, division, print_function
+
+from unittest import TestCase, main
+import numpy as np
+import numpy.testing as npt
+from skbio.stats.composition import (closure, multiplicative_replacement,
+                                     perturb, perturb_inv, power,
+                                     clr, centralize)
+
+
+class CompositionTests(TestCase):
```

Comment:
Another test case that'd be great to add is a check to ensure that the input array(s) aren't modified in-place by the function. In a lot of cases, this won't happen because a copy of the array will be made during execution (e.g., in `np.atleast_2d` or `astype`). However, if the input array is a numpy array of the correct `dtype` already, there's a chance that the manipulations are applied to the input instead of a copy.

I don't think any of the functions modify the input in-place, but it'd be good to test this to ensure this behavior doesn't change in the future.